app/controllers/contact_preferences_controller.rb

```diff
@@ -0,0 +1,31 @@
+class ContactPreferencesController < ApplicationController
+  def show
+    @contact = Contact.find_by(token: token)
+
+    return if @contact
+
+    flash[:notice] = 'Invalid session'
```

Comment:
@despo any reason why this string is not in the locale files?

Reply:
No reason. Happy to move it there!